### PR TITLE
Add a strain test for pinned damage formulation

### PR DESCRIPTION
**Pull request by gassmoeller**
Depends on #14.

Adds a test for the pinned damage formulation in simple shear. I have not yet checked if the numbers mentioned in the test prm comment are correct or expected.

### CODE REVIEW

File: tests/grain_size_strain_pinned.prm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    set Minimum viscosity                = 1e16
+    set Reference temperature            = 1600
+    set Recrystallized grain size        =
+
+    set Grain growth activation energy       = 4e5
+    set Grain growth activation volume       = 0.0
+    set Grain growth rate constant           = 1.92E-010
+    set Grain growth exponent                = 3
+    set Average specific grain boundary energy = 1.0
+    set Work fraction for boundary area change = 0.1
```
Comment (jdannberg) [inline tests/grain_size_strain_pinned.prm]: This parameter is not used anymore. Instead, you can set both the Minimum grain size reduction work fraction and the Maximum grain size reduction work fraction to 0.1.

File: contrib/grain_size/dynamic_grain_size_pinned_state.ipynb [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        "To assess the test results we need to compare the output of the test to the terms given in Eq. 8 of Mulyukova and Bercovici (2018): \n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\text{grain_growth} = \\frac {3\\eta_gk_g\\sqrt h_g^{-q}}{(q R^{q-1})} \\exp(\\frac{-E_g}{R_gT}) \\\\\n",
+        "\\text{grain_size_reduction} = - \\frac{f h_g \\Psi}{(3\\eta_g)} R^2\n",
+        "\\end{equation}\n",
+        "\n",
+        "Here, $\\eta_g$ is phase distribution function and $\\sqrt{h_g} = 2/\\pi$. The shear heating term $\\Psi$ is $2\\dot\\varepsilon\\tau$, where $\\tau = 2 \\eta \\dot\\varepsilon$, therefore, $\\Psi = 4 \\eta \\dot\\varepsilon ^2$.\n",
+        "\n",
+        "We can now for any given shear heating rate $\\Psi$ compute the integrated grain size change rate by adding these two terms and compare them to the results of the ASPECT model, which is exactly what we are doing in this notebook.\n"
```
Comment (alarshi) [inline contrib/grain_size/dynamic_grain_size_pinned_state.ipynb]: Could you explain what two terms we are adding? It is not clear to me.

File: contrib/grain_size/dynamic_grain_size_pinned_state.ipynb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        "phase_distribution      = 0.6*0.4\n",
+        "\n",
+        "hg         = (2/np.pi)*(2/np.pi)\n",
+        "R_times_T  = 8.314*1600\n",
+        "\n",
+        "second_in_year = 3600*24*365.25\n",
+        "strain_rate            = 1e-5/second_in_year/2\n",
+        "\n",
+        "# we run the test model for 1 year, because the output is the change per timestep,\n",
+        "# and because the first ASPECT timestep has a length of 0, the output is 0, which is\n",
```
Comment (alarshi) [inline contrib/grain_size/dynamic_grain_size_pinned_state.ipynb]: Is the grain size 0 at the first time step? I don't follow this comment.

File: source/material_model/grain_size.cc [hunk 4]
```diff
@@ -1639,7 +1638,7 @@ namespace aspect
         }
 
       // These properties are only output properties.
-      if (out.template get_additional_output<SeismicAdditionalOutputs<dim>>() == nullptr)
+      if (use_table_properties && out.template get_additional_output<SeismicAdditionalOutputs<dim>>() == nullptr)
```
Comment (alarshi) [inline source/material_model/grain_size.cc]: Thank you for these changes! Could you open another PR with this commit?

**Review by alarshi — COMMENTED** [review]
Thank you! I am happy to merge it after the comments are addressed.

**Review by gassmoeller — COMMENTED** [review]
Ok, I think I should have fixed all the tests and the documentation by now. You can ignore the focal-dealii-master tester that one is failing because of an incompatibility with this ASPECT version and the development version of deal.II, this has been fixed on ASPECT's main branch already (but is not worth fixing on our fork).
Feel free to take another look and merge if you are happy with the changes.

**Comment by alarshi** [comment]
Thank you for fixing all the tests, it looks good to me!

**Pull request merged**